### PR TITLE
[RISCV] Introduce and use BF16 in Xsfvfwmaccqqq intrinsics

### PR DESCRIPTION
**Pull request by sunshaoce**
BF16 implementation based on @joshua-arch1's https://reviews.llvm.org/D152498
Fixed the incorrect f16 type introduced in https://github.com/llvm/llvm-project/pull/68296

---------

Co-authored-by: Jun Sha (Joshua) <cooper.joshua@linux.alibaba.com>

### CODE REVIEW

**Comment by 4vtomat** [comment]
Good catch！
LGTM, leave final decision to others~

**Review by topperc — CHANGES_REQUESTED** [review]
I believe we need to update `Sema::checkRVVTypeSupport` too

**Review by eopXD — APPROVED** [review]
Sorry I did not do this before you.

The patch looks good, I would say we also need tuples of bf16 for completeness. But I will be adding a suite of intrinsics around BFloat16 (https://github.com/riscv-non-isa/rvv-intrinsic-doc/pull/293) so we can probably do that after this landed too.

I would recommend to mention that vector bfloat16 types is introduced in the title of this PR.

**Comment by eopXD** [comment]
Approval is upon addressing Craig's comment.

File: clang/lib/Sema/SemaChecking.cpp
Comment (topperc) [inline clang/lib/Sema/SemaChecking.cpp:6052]: Does zfbfmin provide vector support?

File: clang/lib/Sema/SemaChecking.cpp
Comment (topperc) [inline clang/lib/Sema/SemaChecking.cpp:6054]: Drop the "experimental-". That's not part of the extension name.

**Review by topperc — APPROVED** [review]
LGTM

**Comment by topperc** [comment]
This patch is unfortunately incorrect because  Zvfbfmin implies Zfbfmin but the SiFive CPUs that implement Xsfvfwmaccqqq do not implement Zfbfmin.

**Pull request merged**
